var/spack/repos/builtin/packages/fenics-dolfin/package.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class FenicsDolfin(CMakePackage):
+    """DOLFIN is the C++/Python interface of FEniCS."""
+
+    homepage = "http://fenicsproject.org/"
+    git      = "https://bitbucket.org/fenics-project/dolfin.git"
+
+    # Use this url for version >= 2017.1.0
+    url      = "https://bitbucket.org/fenics-project/dolfin/get/2018.1.0.post2.tar.gz"
+
+    # version('develop', branch='master')
+    version('2018.1.0.post2', sha256='a71db38740a7ea508f8a725af4b08ccd024168b450033b032b003a5aac1708cf')
+    version('2018.1.0.post1', sha256='ca8f95d03cafd999fe592941c2776474130406a09abf072ea2e6e93e249af38b')
+    version('2018.1.0.post0', sha256='6c5195c9795decc021ffc713614b7885098912fcd89e3da79fd8428cbaf8212b')
+    version('2018.1.0',       sha256='2afb54e2f8a2c7be5a89e4ef224b68ae514c32168c9a94b1921d62339f8decd4')
+    version('2017.2.0.post0', sha256='4169c7e4d22af76bad39c2c20443c4fa75356078d4e7360d4ea924c76a5c9cd3')
+    version('2017.2.0',       sha256='90f77796372eed63f529bafa7c05afa3d5bfeb5f378d3e4e9d53959c0c06bbe7')
+    version('2017.1.0.post0', sha256='25e557491d7fdff0967ef99c678ec77eb0e5f58a3a6b8b31c45e5ca2bdc85912')
+    version('2017.1.0',       sha256='a496574e9a1310806838c7ef32d442ef77379f879ab6a5742bbe521f171d5f88')
+
+    # FIXME: Older versions prepend name to version in url
+    # url      = "https://bitbucket.org/fenics-project/dolfin/get/dolfin-2016.2.0.tar.gz"
```

Comment:
Use a `url_for_version` function to support this. Let me know if you need me to point you to an example.

Reply:
Thank you @adamjstewart - that is a neat solution! I found some examples and I'll fix these urls.